src/codeGenerator/evaluator.ts

```diff
@@ -434,22 +434,29 @@ export class Evaluator implements ExpressionVisitor {
             fnValue = Store.FunctionValue.onLoad(fnName, commands);
         } else if (astNode.trigger instanceof Tick) {
             fnValue = Store.FunctionValue.onTick(fnName, commands);
+
+            // Generate an advancement for tick
+            advName = this.storeAdvancementFromTrigger(advName, fnName, astNode, env);
         } else {
             fnValue = Store.FunctionValue.regular(fnName, commands);
 
-            // For things that aren't load or tick, generate an advancement too
-            advName = `.adv${fnName}`;
-            let triggers = this.parseTrigger(astNode.trigger, env);
-            let advValue = new Store.AdvancementValue(
-                // everything is undefined, what has the world come to?? -JL
-                advName, undefined, undefined, undefined, undefined, undefined, fnName, triggers
-            )
-            this.updateStore(advName, advValue, astNode);
+            // For things that aren't load, generate an advancement too
+            advName = this.storeAdvancementFromTrigger(advName, fnName, astNode, env);
         }
         this.updateStore(fnName, fnValue, astNode);
         return advName;
     }
 
+    private storeAdvancementFromTrigger(advName: string, fnName: string, astNode: On, env: EvaluatorEnv) {
+        advName = `.adv${fnName}`;
+        let triggers = this.parseTrigger(astNode.trigger, env);
+        let advValue = new Store.AdvancementValue(
```

Comment:
Best practices suggest that these should be `const` but I'm not going to insist on this change; just FYI

Reply:
i was lazy and just did the `refactor > extract` without checking, but addressed, thanks